jupyter_dashboards/nbextension/notebook/dashboard-view/dashboard-metadata-compatibility.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+/**
+ * This module provides an API to manage compatibility between
+ * Jupyter Dashboards extension metadata.
+ *
+ * Jupyter Dashboard metadata structure:
+ *    https://github.com/jupyter-incubator/dashboards/wiki/Dashboard-Metadata-and-Rendering
+ */
+define([
+    'jquery',
+    'base/js/namespace',
+    './object-util'
+], function(
+    $,
+    IPython,
+    ObjectUtil
+) {
+    'use strict';
+
+    var GRID_DEFAULT = 'grid_default';
+    var REPORT_DEFAULT = 'report_default';
+
+    // Mapping of view type to view id (from dashboard-metadata.js)
+    // This will change when multiple views of the same type are supported.
+    var VIEW_TO_ID = {
+        grid: GRID_DEFAULT,
+        report: REPORT_DEFAULT
+    };
+
+    return {
+        /**
+         * Converts version 0.x metadata structure to version 1.
+         * Versions 0.x and 1 currently do not overlap.
+         * This must be run AFTER the version 1 metadata is initialized in
+         * `dashboard-metadata.js`.
+         */
+        convert: function() {
```

Comment:
If we're going to play the up-convert game, I think we should future proof a bit. Should we make this more amenable to v0 -> v1 conversion explicitly so that later, another convert function / delegate can hand v1 -> v2 and so on?